### PR TITLE
fix(rum-vue): vue 3 support

### LINKED ISSUE #1058: Vue 3
does this work with Vue 3? if i try like in the example i get an error
```
cannot read property use of undefined
```

### PR DESCRIPTION
**Pull request by mdschweda**
This change introduces Vue 3 support / fixes an error when loading the plugin in Vue 3.

See also https://v3.vuejs.org/guide/migration/global-api.html#vue-prototype-replaced-by-config-globalproperties

Tested on an ECK staging environment with a real world app.

Closes #1058

### CODE REVIEW

**Comment by mdschweda** [comment]
Without means to investigate the npm log I consider the CI pipeline error to be out of scope of this PR. Apparently, error 137 indicates insufficient RAM on the test node. I will remove the semicolon that causes a prettier warning in a last commit --amend if there are no further requests from your side

**Comment by alfkonee** [comment]
@vigneshshanmugam  Please any Progress on this

**Comment by mdschweda** [comment]
@alfkonee There are still problems with the e2e tests. I can't access the logs. They run on saucelabs and I don't feel like paying 200 bucks to finish the PR or make changes blindly and ask Vignesh to manually hit the pipeline button for me on the off chance to fix the problem by accident. I'll try to get the tests to run locally when I find some time even though that was a nightmare last time I tried.

**Review by alfkonee — CHANGES_REQUESTED** [review]
@mdschweda  I'd suggest you use `JSON.stringify()` on the reason object for the affected to help make the reason clearer to read in the logs at the foloowing positions
in 
\packages\rum-core\test\performance-monitoring\performance-monitoring.spec.js:612
```suggestion 
-            `Failed sending http-request transaction to the server, reason: ${reason}`
+           `Failed sending http-request transaction to the server, reason: ${JSON.stringify(reason)}`
```
\packages\rum-core\test\performance-monitoring\performance-monitoring.spec.js:190
```suggestion 
-'Failed sending page load metrics so the server, reason: ' +
-                reason
+'Failed sending page load metrics so the server, reason: ' +
+                JSON.stringify(reason)
```

File: packages/rum-vue/package.json
Comment (vigneshshanmugam) [inline packages/rum-vue/package.json:3]: will publish it as 2.0.0 as its breaking change and also to keep 1.x focussed on prior vue versions.

**Review by vigneshshanmugam — COMMENTED** [review]
Thanks @mdschweda for the PR, Looks good to me. 

I will push some changes
- NPM 6 stuff
- Package version change

Should be good to go after that. Apologies for the delay in getting this in.

File: packages/rum-vue/package.json
Comment (vigneshshanmugam) [reply packages/rum-vue/package.json]: Switched to 1.3.1 as before, will be changed by our release script when doing major bump.

**Comment by vigneshshanmugam** [comment]
Vue 3 dropped support for IE 11 and webpack build is timing out. Have to change the browser list for vue package and change to dev mode to fix the build. Will take care of it from my side.

**Comment by vigneshshanmugam** [comment]
Waiting for the CI issue to get fixed - #1111

**Review by vigneshshanmugam — APPROVED** [review]
LGTM. Thanks @mdschweda, Here are the next steps

- I will update the docs to reflect 1.x support for vue 2 and 2.x for vue 3
- Will do a release for 2.0.0 for `@elastic/apm-rum-vue`

**Pull request merged**
